Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -158,6 +161,12 @@ public enum TypeDescription: Codable, Hashable {
       let typeDescriptions = try values.decode([Self].self, forKey: .typeDescriptions)
       self = .tuple(typeDescriptions)
 
+    case Self.closureDescription:
+      let typeDescriptions = try values.decode([Self].self, forKey: .typeDescriptions)
+      let text = try values.decodeIfPresent(String.self, forKey: .text)
+      let typeDescription = try values.decode(Self.self, forKey: .typeDescription)
```

Comment:
Thoughts on re-using these existing keys? It's a little hacky, but it does work.

Biggest benefit of introducing new keys is to create a key that records a `Bool` rather than `String`.

Reply:
I had a question above about `@escaping`. How we handle that could impact how we approach new keys.

A key for boolean doesn't sit right with me: we'd be creating a general-purpose key for a single existing use case. It's not clear to me that we'd use the key again.

If we are going to create a new key I'd recommend we create one that is specific to this use case. I don't think there is much downside to more semantically meaningful keys.

I also wonder if we could use the `caseDescription` to decode sub-objects that are strongly typed.

```swift
enum CodingKeys: String, CodingKey {
  case caseDescription
  case typeData
}
```

Then we would define a simple `Codable` struct for each object with the properties we want, and store the encoded data for `typeData` key.

So for the closure case we could imagine defining a struct like this.

```swift
struct Closure: Codable {
  let parameterTypes: [TypeDescription]
  let returnType: [TypeDescription]
  let isEscaping: Bool
  let doesThrow: Bool
}
```

`TypeDescription` is becoming complex enough that the additional structure could be valuable. We would need to understand if there is a major increase in decoding/encoding time.

Reply:
> A key for boolean doesn't sit right with me: we'd be creating a general-purpose key for a single existing use case. It's not clear to me that we'd use the key again.

Entirely fair. I agree with you the creating specific keys for our use case would be a reasonable approach.

I like the idea of creating `Codable` structs for each enum case and using those to serialize/deserialize, but that feels out of scope for this PR. @bachand can you create an issue for this? Seems like a good improvement that we could tackle in one fell swoop. I don't expect using strongly-typed `Codable` structs would negatively impact performance.

Only downside I can think of to that change is that it'd technically be a breaking change. But hey, the repo issue still in beta so nbd.

Reply:
Used new keys in 728b500

Reply:
I fully agree with @bachand about the usage of a struct here and moving away from the enum.

Reply:
https://github.com/fdiaz/SwiftInspector/issues/109